src/app/pages/my-data-specification-detail/my-data-specification-detail.component.ts

```diff
@@ -372,6 +383,50 @@ export class MyDataSpecificationDetailComponent implements OnInit, OnDestroy {
       });
   }
 
+  shareDataSpecification() {
+    if (!this.dataSpecification || !this.dataSpecification.id) {
+      return;
+    }
+
+    this.dataSpecificationService
+      .shareWithDialog(this.dataSpecification.readableByAuthenticatedUsers as boolean)
+      .subscribe((response: ShareDataSpecificationDialogResponse) => {
```

Comment:
There are inner `subscribe()` calls here, is there a better way of chaining together the observable calls after the dialog closes?